idris glfw failing to link/run

## Issue description

Running the idris-glfw example at https://github.com/eckart/glfw-idris/tree/master/example/hello fails on nix.

```
[nix-shell:~/glfw-idris/example/hello]$ make
idris --install example.ipkg
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwDestroyWindow':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwDestroyWindow+0x32): undefined reference to `glfwDestroyWindow'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwInit':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwInit+0x29): undefined reference to `glfwInit'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwMakeContextCurrent':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwMakeContextCurrent+0x32): undefined reference to `glfwMakeContextCurrent'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwPollEvents':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwPollEvents+0x27): undefined reference to `glfwPollEvents'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwTerminate':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwTerminate+0x27): undefined reference to `glfwTerminate'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwWindowShouldClose':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwWindowShouldClose+0x34): undefined reference to `glfwWindowShouldClose'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwCreateWindow':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwCreateWindow+0xcb): undefined reference to `glfwCreateWindow'
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwCreateWindow+0x1c7): undefined reference to `glfwCreateWindow'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwGetFunctionKey':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwGetFunctionKey+0x97): undefined reference to `glfwGetKey'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwSetInputMode':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwSetInputMode+0x9e): undefined reference to `glfwSetInputMode'
/run/user/1001/ccsCavWg.o: In function `_idris_Graphics_46_Util_46_Glfw_46_glfwWindowHint':
idris29951-0.c:(.text._idris_Graphics_46_Util_46_Glfw_46_glfwWindowHint+0x97): undefined reference to `glfwWindowHint'
collect2: error: ld returned 1 exit status
FAILURE: "/nix/store/isg8rxaxkijl9x3hr2gzsf8pqfnqxg3k-gcc-wrapper-7.4.0/bin/gcc" ["-fwrapv","-fno-strict-overflow","-std=c99","-pipe","-fdata-sections","-ffunction-sections","-D_POSIX_C_SOURCE=200809L","-DHAS_PTHREAD","-DIDRIS_ENABLE_STATS","-I.","-Wl,-gc-sections","/run/user/1001/idris29951-0.c","-L/nix/store/mdbpr32kzamgayyvfr5wq2qc5cwmkv90-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts","-lidris_rts","-lgmp","-DIDRIS_GMP","-lpthread","-I/nix/store/mdbpr32kzamgayyvfr5wq2qc5cwmkv90-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts","-lm","-I.","-I/nix/store/nhrb649m7wjsljbc2cxx3w9gpw4iprbz-idris-1.3.1/libs/prelude","-I/nix/store/nhrb649m7wjsljbc2cxx3w9gpw4iprbz-idris-1.3.1/libs/base","-I/nix/store/nhrb649m7wjsljbc2cxx3w9gpw4iprbz-idris-1.3.1/libs/effects","-I/nix/store/nhrb649m7wjsljbc2cxx3w9gpw4iprbz-idris-1.3.1/libs/glfw","-I/nix/store/jiai43fv8pq2v4ybz18363yxwh8cqgkk-idris-1.3.1/libs/prelude","-I/nix/store/jiai43fv8pq2v4ybz18363yxwh8cqgkk-idris-1.3.1/libs/base","-I/nix/store/jiai43fv8pq2v4ybz18363yxwh8cqgkk-idris-1.3.1/libs/effects","-I/nix/store/jiai43fv8pq2v4ybz18363yxwh8cqgkk-idris-1.3.1/libs/glfw","-I/nix/store/sb4vzx6daid29198l2n63lxbz8vl27kj-idris-1.3.1/libs/prelude","-I/nix/store/sb4vzx6daid29198l2n63lxbz8vl27kj-idris-1.3.1/libs/base","-I/nix/store/sb4vzx6daid29198l2n63lxbz8vl27kj-idris-1.3.1/libs/effects","-I.","-I/nix/store/gpalrwhq5pdbidsa1r9563nrsfm9zrrr-idris-1.3.1/libs/prelude","-I/nix/store/gpalrwhq5pdbidsa1r9563nrsfm9zrrr-idris-1.3.1/libs/base","-o","/home/lc/glfw-idris/example/hello/hello"]
```

### Steps to reproduce

    nix-shell -p 'idrisPackages.with-packages (with idrisPackages; [ contrib glfw  ])' -p gmp -p glfw

Followed by trying to build the example at https://github.com/eckart/glfw-idris/tree/master/example/hello

--- a/pkgs/development/idris-modules/glfw.nix
+++ b/pkgs/development/idris-modules/glfw.nix
@@ -10,7 +10,13 @@ build-idris-package  {
 
   idrisDeps = [ effects ];
 
+  nativeBuildInputs = [ pkgs.pkgconfig ];
   extraBuildInputs = [ pkgs.glfw ];
+
+  postPatch = ''
+    substituteInPlace src/MakefileGlfw \
+      --replace glfw3 "glfw3 gl"
+  '';
 
   src = fetchFromGitHub {
     owner = "eckart";

